### ISSUE TITLE
Teach an e-commerce search engine to better understand its users

### ISSUE THREAD

**Issue by dnaranjo89** [dnaranjo89/test_script#30]
# Title
Teach an e-commerce search engine to better understand its users
---
# Metadata
Tags: ML, neural networks, deep learning, elastic, elasticsearch, pyspark
Audience Level: All
---
# Abstract
In this talk we will present NLP and ML techniques for e-commerce query understanding using word embeddings techniques. We will also detail why  Python is the perfect language to implement both off-line and on-line components, leveraging the rich ecosystem of scientific and engineering libraries.
---
# Description
Online retailer and E-commerce search engines usually rely on human curated metadata and taxonomies to help users find the products they are looking for. In this talk we will present how NLP and ML techniques can be leveraged to close the gap between the users’ noisy language and a curated product catalogue, focusing on the applications of spell checking and intent classification.

We will present two applications using a mix of shallow and deep learning techniques for NLP: 

* Training statistical spellchecker traditionally required an annotated dataset of misspelled sentences and their corrections. Instead, web companies have relied on massive query logs and used iterative procedures. We will showcase a simple approach where we complement Elastic phrase suggester with an approximated error model based on the distributional hypothesis and trained using the word2vec algorithm with the Gensim and Fasttext librarires. 
* Inferring the intent of a user despite a noisy or ambiguous search query can be used both to complement the traditional content based search and to prompt users to disambiguate their meaning. We will present how to build an intent classifier based on sentence embeddings with the FastText library and a more sophisticated metric learning model implemented in PyTorch, attaching noisy user queries to a pre-existing taxonomy of concepts.
---
# Notes
 * I am a ML engineer embedded in the team responsible for argos.co.uk search engine and as such I am involve in building data driven features from the early modelling and data pre-processing in Pyspark down to deploying Python micro-services responsible of real time serving of user queries.
 * I love the Python language for its versatility, and its ability to write code fulfilling best engineering practices while drawing from the immense wealth of scientific libraries.
---
# Speaker Bio
Guillaume has a PhD in applied mathematics and statistics and has learned software engineering on the job while building recommender and NLP systems in the industry. He is passionate about applying both scientific and agile approaches to deploy ML systems in the wild world. He has recently joined Argos where he is taking part in the digital transformation of this traditional retailer into a tech driven company.
